src/pal/src/file/file.cpp

```diff
@@ -738,7 +775,21 @@ CorUnix::InternalCreateFile(
         goto done;
     }
 
-    if (strcpy_s(pLocalData->unix_filename, sizeof(pLocalData->unix_filename), lpUnixPath) != SAFECRT_SUCCESS)
+    if (pLocalData->unix_filename != NULL)
```

Comment:
`_ASSERTE(pImmutableData->lpFileName == NULL)` without the error code path would be enough here, this condition would indicate a bug in the PAL and cannot be triggered by incorrect arguments passed to the PAL functions.